lib/modules/helpers.js

```diff
@@ -1,9 +1,12 @@
-function generateValidLink(url, username = '') {
+function generateValidLink(url, username) {
   let validLink = url
-  const validRegex = /^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&\/=]*)$/
+  let validUsername = username || 'all-contributors'
 
-  if (!validLink.startsWith("http")) validLink = `http://${url}/`
-  if (!validLink.match(validRegex)) validLink = `https://github.com/${username}/`
+  const validRegex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/
+  const validRegexWithoutProtocol = /^[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/
+
+  if (validLink.match(validRegexWithoutProtocol)) validLink = `http://${url}/`
+  if (!validLink.match(validRegex)) validLink = `https://github.com/${validUsername}/`
```

Comment:
One suggestion here, this is basically creating `html_url` that was already in the calling function. If how we call this function is changed like

```js
getUserLink(blog, html_url)
```

This function can be changed into

```js
function getUserLink(blog, githubProfileURL) {
	const validRegexWithScheme = /same as now/
	const validRegexWithoutScheme = /same as now/
	if (validRegexWithProtocol.test(blog)) return blog;
	if (validRegexWithoutProtocol.test(blog)) return `http://${blog}`;
	return githubProfileURL
}

```
This way, someone reading the code in the future only have to think about which `return` would be triggered. 
What do you think?